client/src/components/pages/Players.js

```diff
@@ -1,289 +1,303 @@
+      message.error("Something went wrong, failed to delete team.");
+    }
   };
 
-  handleModeChange = (e) => {
+  const handleModeChange = (e) => {
     if (e.key === "players") {
-      this.setState(
-        {
-          mode: "players",
-          sort: "rank",
-        },
-        () => this.sortPlayers("rank")
-      );
+      setMode("players");
+      setSort("rank");
+      setPlayers(sortedPlayers("rank", players));
     } else {
-      console.log("sort team");
-      this.setState(
-        {
-          mode: "teams",
-          sort: "alpha",
-        },
-        () => this.sortTeams("alpha")
-      );
+      setMode("teams");
+      setSort("alpha");
+      setTeams(sortedTeams("alpha", teams));
     }
   };
 
-  onFinish = async (formData) => {
-    const players = formData.players.split(",").map((s) => s.trim());
+  const onFinish = async (formData) => {
+    setLoading(true);
 
     try {
-      const team = await post("/api/team", {
-        players: players,
+      const teamData = await post("/api/team", {
+        players: formData.players,
         name: formData.name,
-        tourney: this.props.tourney,
+        tourney: tourney,
         icon: formData.icon,
       });
-      this.setState((state) => ({
-        teams: [...state.teams, team],
-      }));
+
+      setTeams(sortedTeams(sort, [...teams, teamData]));
+      setLoading(false);
       message.success(`Added team ${formData.name}`);
     } catch (e) {
       message.error("Couldn't create this team. Make sure all team members are spelled correctly");
     }
   };
 
-  handleTeamEdit = async (team) => {
-    this.setState({ loading: true });
-    const _id = this.state.editingTeam;
+  const handleTeamEdit = async (team) => {
+    setLoading(true);
+    const _id = editingTeam;
+
     try {
       const newTeam = await post("/api/edit-team", {
         ...team,
         _id,
-        tourney: this.props.tourney,
+        tourney: tourney,
       });
-      this.setState((state) => ({
-        teams: state.teams.map((t) => {
+
+      setTeams(
+        teams.map((t) => {
           if (t._id === _id) return newTeam;
+
           return t;
-        }),
-      }));
+        })
+      );
+      message.success("Team edited succesfully!");
     } catch (e) {
-      alert(`Couldn't update team: ${e}`);
+      message.error(`Couldn't update team: ${e}`);
     }
-    this.setState({
-      loading: false,
-      editingTeam: -1,
-    });
+
+    setLoading(false);
+    setEditingTeam(-1);
   };
 
-  handleTeamEditStats = async (formData, _id) => {
-    const newTeam = await post("/api/team-stats", {
-      ...formData,
-      _id,
-      tourney: this.props.tourney,
-    });
-    this.setState((state) => ({
-      teams: state.teams.map((t) => {
+  const handleTeamEditStats = async (formData, _id) => {
+    try {
+      const newTeam = await post("/api/team-stats", {
+        ...formData,
+        _id,
+        tourney: tourney,
+      });
+
+      setTeams((t) => {
```

Comment:
this isn't quite right -- i noticed that it fails to update the teams
it should be like this:
```js
setTeams(teams.map((t) => {
```